### ISSUE TITLE
When headless='virtual' is enabled, Xvfb does not terminate automatically, causing threads not to exit

### ISSUE THREAD

**Issue by Nongzhsh** [daijro/camoufox#49]
### Describe the bug:

When `headless='virtual'` is enabled in the Camoufox library, Xvfb does not terminate automatically at the end of the program. This behavior prevents the thread from exiting properly.

### To Reproduce:

``` python
import threading
from concurrent.futures import ThreadPoolExecutor
from camoufox import Camoufox

def run_job(url):
    with Camoufox(
            headless='virtual',
            humanize=True,
            block_webrtc=True,
            block_images=False, ) as browser:
        page = browser.new_page()
        page.goto(url)
        page.wait_for_timeout(6000)
        page.screenshot(path=f"{threading.current_thread().name}.png")
        page.close()

if __name__ == "__main__":
    with ThreadPoolExecutor(max_workers=3) as executor:
        for index in range(5):
            executor.submit(run_job, f"https://bounty-nodejs.datashield.co/scraping/{index}")
```

### Version:
OS:                 WSL2 (Ubuntu 22.04.4 LTS)
Pip package:    v0.3.0
Camoufox:       v130.0.1-beta.13 (Up to date!)

**Comment by Nongzhsh** [comment]
`os.system("pkill -u $(whoami) Xvfb")` works for me after all futures are completed. 
However, I feel like this isn't a good idea. 😂

**Comment by daijro** [comment]
Fixed in 0.3.1!

**Comment by Nongzhsh** [comment]
> Added an experimental fix in 0.3.1. 👍
> 
> Let me know how it works for you.


`version 0.3.2 ` works great!
I’ve tested it, much appreciated! 👍

**Comment by Nongzhsh** [comment]
Hello, @daijro ! The Xvfb auto-termination fix worked great, but I’m noticing a new issue when running multiple Camoufox instances at the same time. If I launch the example script in different shell sessions concurrently, the browser takes a really long time to start up successfully. I wonder if there’s some kind of resource contention happening, maybe around the `DISPLAY `environment variable? Just a guess (I’m not super familiar with Python), but this might be a bug. 😅
